src/app/_helpers/filter-conf.ts

```diff
@@ -0,0 +1,68 @@
+import { FilterExecution } from '../_models/filterExecution';
+
+export const filterConf: FilterExecution[] = [
+  {
+    label: 'workflow',
+    name: 'WORKFLOW',
+    multi: true,
+    options: [
+      { value: 'enrich' },
+      { value: 'import a' },
+      { value: 'import b' },
+      { value: 'normalise' },
+      { value: 'preview' },
+      { value: 'validate' },
+      { value: 'transform' },
+    ],
+  },
+  {
+    label: 'date',
+    name: 'DATE',
+    options: [
+      { value: 'last 24' },
+      { value: 'last week' },
+      { value: 'last month' },
+      {
+        value: 'from',
+        input: {
+          id: 'date-from',
+          type: 'date',
+          fnOnSet: (val: string): void => {
+            const dateTo = document.querySelector('#date-to');
```

Comment:
ViewChild would require a component per input. ViewChildren could be used, but it would complicate things.  One complication would be the need to use a variant of the technique used in executions-grid to maintain the sibling relationship of the cells in the "rows" (which can't have a wrapping element).  Another complication would be the amount of data having to be passed back and forth between parent and child components in order to relate components (i.e. make one date affect the other).  Using ViewChildren I'd still have to run a selector to filter the QueryList it would return to obtain a reference to the element.

Reply:
OK, I revisited this and Angularified it.  The options now bind to a shared data model.  The interface FilterExecutionProvider was introduced to avoid circular dependencies.  There are no longer any references to document.querySelector